packages/ecs-agent/0002-bottlerocket-default-filesystem-locations.patch

```diff
@@ -0,0 +1,82 @@
+From 98da69b9412ad38c7e7448deb6439dcb04db175b Mon Sep 17 00:00:00 2001
+From: Samuel Karp <skarp@amazon.com>
+Date: Wed, 8 Jul 2020 11:18:35 -0700
+Subject: [PATCH 1/2] bottlerocket: default filesystem locations
+
+---
+ agent/config/config.go      | 6 +++---
+ agent/config/config_unix.go | 4 ++--
+ agent/utils/license.go      | 6 +++---
+ 3 files changed, 8 insertions(+), 8 deletions(-)
+
+diff --git a/agent/config/config.go b/agent/config/config.go
+index 1079684b..b11c0153 100644
+--- a/agent/config/config.go
++++ b/agent/config/config.go
+@@ -51,7 +51,7 @@ const (
+ 	AgentPrometheusExpositionPort = 51680
+ 
+ 	// defaultConfigFileName is the default (json-formatted) config file
+-	defaultConfigFileName = "/etc/ecs_container_agent/config.json"
++	defaultConfigFileName = "/etc/ecs/ecs.config.json"
+ 
+ 	// DefaultClusterName is the name of the default cluster.
+ 	DefaultClusterName = "default"
+@@ -115,13 +115,13 @@ const (
+ 	minimumNumImagesToDeletePerCycle = 1
+ 
+ 	// defaultCNIPluginsPath is the default path where cni binaries are located
+-	defaultCNIPluginsPath = "/amazon-ecs-cni-plugins"
++	defaultCNIPluginsPath = "/usr/lib/amazon-ecs-agent/cni-plugins"
+ 
+ 	// DefaultMinSupportedCNIVersion denotes the minimum version of cni spec required
+ 	DefaultMinSupportedCNIVersion = "0.3.0"
+ 
+ 	// pauseContainerTarball is the path to the pause container tarball
+-	pauseContainerTarballPath = "/images/amazon-ecs-pause.tar"
++	pauseContainerTarballPath = "/usr/lib/amazon-ecs-agent/amazon-ecs-pause.tar"
+ 
+ 	// DefaultTaskMetadataSteadyStateRate is set as 40. This is arrived from our benchmarking
+ 	// results where task endpoint can handle 4000 rps effectively. Here, 100 containers
+diff --git a/agent/config/config_unix.go b/agent/config/config_unix.go
+index ac5f5d0f..1021d916 100644
+--- a/agent/config/config_unix.go
++++ b/agent/config/config_unix.go
+@@ -27,7 +27,7 @@ const (
+ 	// AgentCredentialsAddress is used to serve the credentials for tasks.
+ 	AgentCredentialsAddress = "" // this is left blank right now for net=bridge
+ 	// defaultAuditLogFile specifies the default audit log filename
+-	defaultCredentialsAuditLogFile = "/log/audit.log"
++	defaultCredentialsAuditLogFile = "/var/log/ecs/audit.log"
+ 	// DefaultTaskCgroupPrefix is default cgroup prefix for ECS tasks
+ 	DefaultTaskCgroupPrefix = "/ecs"
+ 
+@@ -48,7 +48,7 @@ func DefaultConfig() Config {
+ 		DockerEndpoint:                      "unix:///var/run/docker.sock",
+ 		ReservedPorts:                       []uint16{SSHPort, DockerReservedPort, DockerReservedSSLPort, AgentIntrospectionPort, AgentCredentialsPort},
+ 		ReservedPortsUDP:                    []uint16{},
+-		DataDir:                             "/data/",
++		DataDir:                             "/var/lib/ecs",
```

Comment:
the path `/data/` was normally to map to path `/var/lib/ecs/data` on the host when agent is running in container, so this actually changes the data dir on the host from `/var/lib/ecs/data` to `/var/lib/ecs`. is it intentional? some feature, e.g. firelens, is assuming that the files are stored in the data folder (e.g. [here](https://github.com/aws/amazon-ecs-agent/blob/master/agent/api/task/task.go#L107)) so changing it here will force the path to diverge in the feature code path as well

Reply:
I can change it to `/var/lib/ecs/data` if desired.  However, it seems like a bug that the Firelens code will assume that to be the path rather than reading the path from the config.

Reply:
Well, the agent already reads the host path from config.DataDirOnHost, which does not contain the "data" folder so we need to append "data/" to it. The complication is caused by the fact that config.DataDirOnHost does not contain folder "data" but config.DataDir does.

Reply:
Yeah, the current code is coupled to that assumption.  To me that seems like a bug in the design; I'll see if I can come up with some suggestions for making it nicer once I get to the bits that need it.

Right now, with `config.DataDirOnHost` set to `/var/lib/ecs`, the Firelens bind ends up being `/var/lib/ecs/data/firelens/%s/config/fluent.conf:/fluentd/etc/fluent.conf`, which seems reasonable.  I don't think it's accessing `ecs_agent_state.json`, so it doesn't matter whether it's in the same folder (`/var/lib/ecs`) or not (`/var/lib/ecs/data`).

Reply:
Done